governance/microservice.go

```diff
@@ -132,18 +153,29 @@ func (w *GovernanceWorker) StartMicroservice(ms_key string, agreementId string,
 							fn = fmt.Sprintf("%v.pem", key)
 						}
 
-						api.UploadPublicKey(fn, []byte(content), w.Config, errHandler(fn))
+						// Keys for different services might have the same key name like service.public.pem so prepend something unique like the agreement id 
+						// but then we have to make sure we delete the key when done with it
+						prepend_string := agreementId + "_" + strconv.Itoa(num_signing_keys) + "_"
```

Comment:
StartMicroservice get called with agreementId as an empty string in some cases.  If agreementId is empty, please use ms_key instead.

Reply:
changed per comment